### PR TITLE
enhance string util functions

### PR DESCRIPTION
**Pull request by fs-eire**
### Description
- make `MakeString` force inline
- refactor ORT_FORCEINLINE macro - move to one place to avoid macro redefinition error
- ~~add a `StringJoin` utility~~

### CODE REVIEW

**Review by snnn — APPROVED** [review]

**Comment by pranavsharma** [comment]
What's the need for writing a wrapper for ostringstream? ostringstream is already general enough.

**Comment by fs-eire** [comment]
> What's the need for writing a wrapper for ostringstream? ostringstream is already general enough.

Using a function call for appending to ostream is helpful when using with macros or other template functions.

`MakeString` is already used in macros like `ORT_ENFORCE()`. `StringJoin` is also going to be useful in WebGPU EP that I am currently working on.

File: include/onnxruntime/core/common/string_join.h [hunk 1]
```diff
@@ -0,0 +1,71 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```
Comment (snnn) [inline include/onnxruntime/core/common/string_join.h:1]: Can you use absl::StrJoin instead?

Comment (fs-eire) [reply include/onnxruntime/core/common/string_join.h:1]: Sure. let me remove string_join.h

**Review by pranavsharma — APPROVED** [review]

**Review by guschmue — APPROVED** [review]

**Pull request merged**